stdlib/asm/collections/sorted_array.masm

```diff
@@ -281,3 +369,38 @@ export.find_key_value
         end
     end
 end
+
+#! Loads a key from the pointer.
+#!
+#! If use_half_key is 1, the key is loaded with the two least significant elements set to zero,
+#! e.g.: [key3, key2, 0, 0], while if it is 0, the full key [key3, key2, key1, key0] is loaded.
+#!
+#! This procedure avoids branches as an optimization.
+#!
+#! Inputs:  [use_half_key, key_ptr]
+#! Outputs: [KEY]
+proc.load_key
```

Comment:
If we invert the meaning of `use_half_key` - i.e., change it to `use_full_key` - then, I think this could be done much simpler - e.g. something like:
```
#! Inputs:  [use_full_key, key_ptr]
#! Outputs: [KEY]
proc.load_key
    padw movup.5 mem_loadw
    # => [FULL_KEY, use_full_key]

    # multiple the two least significant elements by use_full_key
    movup.3 dup.4 mul
    movup.3 movup.4 mul
    movup.3 movup.3
    # => [KEY]
end
```

This should take 14 cycles (but would be good to double-check).

Reply:
Nice optimization, thank you! Refactored `use_half_key` to `use_full_key` and addressed your other comments as well.